pkg/athena/api/api.go

```diff
@@ -26,7 +26,7 @@ func New(sessionCache *awsds.SessionCache, settings sqlModels.Settings) (api.AWS
 	athenaSettings := settings.(*models.AthenaDataSourceSettings)
 
 	httpClientProvider := sdkhttpclient.NewProvider()
-	httpClientOptions, err := athenaSettings.Config.HTTPClientOptions()
+	httpClientOptions, err := athenaSettings.Config.HTTPClientOptions(context.TODO())
```

Comment:
Is there a way for us to pass a context into New to avoid using a `context.TODO()` here?

Reply:
The signature of `New` needs to satisfy the `Loader` definition in grafana-aws-sdk which is what these methods take. 
https://github.com/grafana/grafana-aws-sdk/blob/ab9e896bf1c2d753908ed7f7ed0682b29e8814f6/pkg/sql/api/api.go#L57-L56
https://github.com/grafana/athena-datasource/blob/0ad44f26bb6da681983ca4625a299b5137f20281/pkg/athena/datasource.go#L32-L31

It starts to break quite a few interface definitions along the way, looks like a pretty big refactor. Maybe we can create a ticket to make that available? Let me know if I've missed something!

Reply:
That sounds reasonable! Yeah, I didn't realize it was such an entangled signature